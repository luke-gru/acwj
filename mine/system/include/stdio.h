#ifndef _STDIO_H_
#define _STDIO_H_

#include <stddef.h>
#include <stdarg.h>

#define EOF (-1)

// This FILE definition will do for now
typedef char *FILE;
extern FILE *stderr;
extern FILE *stdout;

FILE *fopen(char *pathname, char *mode);
size_t fread(void *ptr, size_t size, size_t nmemb, FILE *stream);
size_t fwrite(void *ptr, size_t size, size_t nmemb, FILE *stream);
int fclose(FILE *stream);
int printf(char *format);
int fprintf(FILE *stream, char *format);
size_t getline(char **lineptr, size_t *n, FILE *stream);

FILE *popen(const char *command, const char *type);
int pclose(FILE *stream);

int vfprintf(FILE *stream, const char *format, va_list ap);

#endif	// _STDIO_H_
