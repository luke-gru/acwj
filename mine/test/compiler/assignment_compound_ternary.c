#include <stdio.h>

int main(void) {
  int a, b, c, d;
  int e;
  a = 10;
  b = 5;
  c = 2;
  d = 1;
  e = (a > b) ? (a+b) : a;
  printf("e: %d\n", e); // 15
  printf("d: %d\n", d); // 1
  d += ((a > b) ? (a+b) : a);
  printf("a: %d\n", a); // 10
  printf("b: %d\n", b); // 5
  printf("d: %d\n", d); // 16
  d = 2;
  d *= ((a > b) ? (a+b) : a);
  printf("d: %d\n", d); // 30
  return(0);
}
