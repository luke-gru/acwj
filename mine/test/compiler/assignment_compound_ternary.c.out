e: 15
d: 1
a: 10
b: 5
d: 16
d: 30
